vue-app/src/components/ContributionModal.vue

```diff
@@ -166,112 +167,123 @@ export default class ContributionModal extends Vue {
-    // Reload contribution pool size
-    this.$store.dispatch(LOAD_ROUND_INFO)
-    // Vote (step 3)
-    const messages: Message[] = []
-    const encPubKeys: PubKey[] = []
-    let nonce = 1
-    for (const [recipientIndex, voiceCredits] of this.votes) {
-      const [message, encPubKey] = createMessage(
-        contributor.stateIndex,
-        contributor.keypair,
-        null,
-        coordinatorPubKey,
-        recipientIndex,
-        voiceCredits,
-        nonce
+      const voiceCredits = getEventArg(
+        contributionTxReceipt,
+        maci,
+        'SignUp',
+        '_voiceCreditBalance'
       )
-      messages.push(message)
-      encPubKeys.push(encPubKey)
-      nonce += 1
-    }
-    this.step += 1
-    try {
-      await waitForTransaction(
-        fundingRound.submitMessageBatch(
-          messages.reverse().map((msg) => msg.asContractParam()),
-          encPubKeys.reverse().map((key) => key.asContractParam())
-        ),
-        (hash) => (this.voteTxHash = hash)
-      )
-      this.$store.dispatch(SAVE_COMMITTED_CART_DISPATCH)
-    } catch (error) {
-      this.voteTxError = error.message
-      return
+      if (!voiceCredits.mul(voiceCreditFactor).eq(total)) {
+        throw new Error('Incorrect amount of voice credits')
+      }
+      const contributor = {
+        keypair: contributorKeypair,
+        stateIndex: stateIndex.toNumber(),
+      }
+      // Save contributor data to storage
+      this.$store.commit(SET_CONTRIBUTOR, contributor)
+      this.$store.dispatch(SAVE_CONTRIBUTOR_DATA)
+      // Set contribution and update round info
+      this.$store.commit(SET_CONTRIBUTION, total)
+      // Reload contribution pool size
+      this.$store.dispatch(LOAD_ROUND_INFO)
+      // Vote (step 3)
+      const messages: Message[] = []
+      const encPubKeys: PubKey[] = []
+      let nonce = 1
+      for (const [recipientIndex, voiceCredits] of this.votes) {
+        const [message, encPubKey] = createMessage(
+          contributor.stateIndex,
+          contributor.keypair,
+          null,
+          coordinatorPubKey,
+          recipientIndex,
+          voiceCredits,
+          nonce
+        )
+        messages.push(message)
+        encPubKeys.push(encPubKey)
+        nonce += 1
+      }
+      this.step += 1
+      try {
+        await waitForTransaction(
+          fundingRound.submitMessageBatch(
+            messages.reverse().map((msg) => msg.asContractParam()),
+            encPubKeys.reverse().map((key) => key.asContractParam())
+          ),
+          (hash) => (this.voteTxHash = hash)
+        )
+        this.$store.dispatch(SAVE_COMMITTED_CART_DISPATCH)
+      } catch (error) {
+        this.voteTxError = error.message
+        return
+      }
+      this.step += 1
+    } catch (err) {
+      /* eslint-disable-next-line no-console */
+      console.log(err)
+      this.error =
+        'Something unexpected ocurred. Refresh the page and try again.'
```

Comment:
I've added this wrapping try/catch for handling unexpected errors in the entire `contribute` process. Things that are not being caught by the others try/catch that we currently have in each transaction call.

Reply:
Any suggestion about the message is appreciated :)